src/SelfService/Application/KafkaTopicApplicationService.cs

```diff
@@ -99,6 +110,21 @@ List<string> GetRequired(JsonDocument doc)
         // See: https://yokota.blog/2021/03/29/understanding-json-schema-compatibility/
 
 
+        var previousType = GetJsonType(previousSchemaDocument);
+        var newType = GetJsonType(newSchemaDocument);
+        if (previousType != null && newType == null)
+        {
+            throw new InvalidMessageContractRequestException(
+                "Cannot change schema type from " + previousType + " to null"
+            );
+        }
+        if (previousType != null && newType != null && previousType != newType)
```

Comment:
would a single comparison `previousType != newType` not be suficient?

Reply:
it would